### PR TITLE
Split PaletteMap out of Gfx.cpp/Gfx.h

### PR DESCRIPTION
**Pull request by ZehMatt**

### CODE REVIEW

**Comment by LeftofZen** [comment]
Looks like a straight copy-paste of the struct? I'll let the other guys look at it first before I approve it, but it looks good to me.

File: src/OpenLoco/Graphics/PaletteMap.cpp
Comment (duncanspumpkin) [inline src/OpenLoco/Graphics/PaletteMap.cpp:24]: It was pointed out in openrct2 that we never set the initialised variable to true.

Comment (ZehMatt) [reply src/OpenLoco/Graphics/PaletteMap.cpp:24]: I just moved things around and didn't really look for the details, but I can probably improve that part.

File: src/OpenLoco/Graphics/Gfx.cpp
Comment (duncanspumpkin) [inline src/OpenLoco/Graphics/Gfx.cpp:1931]: I don't think this should be moved it's for the actual rgb palette rather than the u8 palette maps

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**
